gradio/components/file_explorer.py

```diff
@@ -202,11 +203,9 @@ def ls(self, subdirectory: list | None = None) -> list[dict[str, str]] | None:
 
         return folders + files
 
-    def _safe_join(self, folders):
-        combined_path = os.path.join(self.root_dir, *folders)
-        absolute_path = os.path.abspath(combined_path)
-        if os.path.commonprefix([self.root_dir, absolute_path]) != os.path.abspath(
-            self.root_dir
-        ):
-            raise ValueError("Attempted to navigate outside of root directory")
-        return absolute_path
+    def _safe_join(self, folders: list[str]):
+        if not folders or len(folders) == 0:
+            return self.root_dir
+        head, tail = folders[0], folders[1:]
```

Comment:
What's the purpose of splitting head and tail? Why can't we do `combined_path = UserProvidedPath(os.path.join(*folders))` directly

Reply:
It was originally like that but then I ran into errors in the functional tests. I thought it was because `os.path.join` couldn't handle a single argument but it's actually because the list was empty.

So long story short, I think we can use the line you're suggesting since I added the empty list check right above.